racket/src/ChezScheme/c/prim5.c

```diff
@@ -878,63 +878,63 @@ static char *s_getwd() {
 static ptr s_set_code_byte(p, n, x) ptr p, n, x; {
     I8 *a;
 
-    S_thread_start_code_write();
+    S_thread_start_code_write(0, static_generation);
```

Comment:
This could be `static_generation-1`. But, also, this seems extremely expensive if code is ever compiled on the flay. Probably the `c-mkcode` loop in "compile.ss" needs to be made non-interruptable and enable/disable writing just once.

Reply:
The `c-mkcode` loop is already in a critical section combined with allocating the code, so now I think this can just be 0.

Reply:
... although it still seems expensive to keep toggling.